libraries/plugins/elasticsearch/include/graphene/elasticsearch/elasticsearch_plugin.hpp

```diff
@@ -174,9 +175,94 @@ struct bulk_struct {
    optional<visitor_struct> additional_data;
 };
 
+struct adaptor_struct {
+   variant adapt(const variant_object& op)
+   {
+       fc::mutable_variant_object o(op);
+       vector<string> keys_to_rename;
+       for (auto i = o.begin(); i != o.end(); ++i)
+       {
+           auto& element = (*i).value();
+           if (element.is_object())
+           {
+               const string& name = (*i).key();
+               auto& vo = element.get_object();
+               if (vo.contains(name.c_str()))
+                   keys_to_rename.emplace_back(name);
+               element = adapt(vo);
+           }
+           else if (element.is_array())
+               adapt(element.get_array());
+       }
+       for (const auto& i : keys_to_rename)
```

Comment:
I am trying to contact Damir from Dascoin to ask some of this questions amd invite to participate here if the can and want. 
By investigating a bit the final data in elastic i found that for example underscore is used at ` operation_history.op_object.amount_.amount` in a transfer operation. Probably in this case there are issues with a field and subfield named the same but not 100% certain about it.

Reply:
in https://pastebin.com/7n2YkkTH for operation_type=17 `amount_` is used. trying to find more samples by checking all the operation types.

Reply:
operation_type 33 https://pastebin.com/Pr1XiWDg  uses _amount but please also note `owner_`

Reply:
Hm, maybe ES gets confused if for a given field name the value can have different types? I. e. either a JSON-object or a number in the case of "amount"? In that case the selection algorithm for keys to rename might not work in all cases.

Reply:
yes, that is correct. elastic creates database schema as it see fields for the first time. the same field cant be of different types or the second record will not be inserted.

Reply:
So the schema ignores the path of a nested field and only looks at the field name?

Reply:
i think the path is not ignored, but for example a portion of an account_update operation will have:
- `operation_history.op_object.owner.address_auths`: this is a nested type of `owner`.

in operation 33:
- `operation_history.op_object.owner` is of type string.

same path, different types.

Reply:
Ok, but I think that will not be caught by the keys_to_rename check. The check only operates within a single operation.

Reply:
You are right here. it seems that the `amount_.amount` is done at this check. There is maybe an issue with having `field.field` with Elastic. Will research more and try to figure this out.

Reply:
Hello folks, maybe I can shed some light here; please note that the adaptor is created to best fit the Dascoin use, so it might need some tweaking for BitShares. 
First thing to note is the fact that ES creates schema as it see fields for the first time, meaning that i.e. if in one operation we see a fields called 'foo' of type String, then all subsequent fields named 'foo' need to be of the same type. That why I had to rename some of the fields.
Second thing is that ES sometimes gets confused with fields named 'foo.foo.bar', in which case I also had to perform a rename.

Reply:
Thanks @damirn . I like the approach and as you mention we have to do a bit more in Bitshares side to make this work.

By checking the console output of elastic while adding data i am seeing:

`[operation_history.op_object.total_claimed.amount] of different type, current_type [text], merged_type [long]`
I think this one is because sometimes numbers came from the core as "212112"(if they are too big) and some others just as 222. so we probably can convert to string all numbers(or all to long) in `amount` fields. Other ideas welcome.

similar issue here:
`mapper [operation_history.op_object.min_to_receive.amount] of different type, current_type [long], merged_type [text]`

Other i am observing to rename in the proposals:

`Can't merge a non object mapping [operation_history.op_object.proposed_ops.op] with an object mapping [operation_history.op_object.proposed_ops.op]`

I propose to add code to fix those in Bitshares, keep going and find more errors, fix if needed. What do you think @pmconrad ?